polaris.shopify.com/pages/[...slug].tsx

```diff
@@ -43,7 +43,14 @@ export const getStaticProps: GetStaticProps<Props, {slug: string[]}> = async ({
   const slug = params?.slug;
   if (!slug)
     throw new Error('Expected params.slug to be defined (as string[])');
-  const mdRelativePath = `${contentDir}/${params.slug.join('/')}/index.md`;
+
+  const slugPath = `${contentDir}/${params.slug.join('/')}`;
+  const pathIsDirectory = fs.existsSync(slugPath)
+    ? fs.lstatSync(slugPath).isDirectory()
+    : false;
```

Comment:
I got a bit confused with the ternary returning false in the "else" case
So I was thinking that maybe this could be written like:

```suggestion
  const pathIsDirectory = fs.existsSync(slugPath) && fs.lstatSync(slugPath).isDirectory();
```

It looks like you need the path to exist and it needs to be a directory, right?

wdyt?

Reply:
Ah, sweet!

I chickened out because I wasn't sure if JavaScript would evaluate all operands and thus throw an error on the second condition. But your code 100% works. MDN:

```
# Short circuit evaluation

The logical AND expression is a short-circuit operator. As each operand is converted to a boolean, if the result of one conversion is found to be false, the AND operator stops and returns the original value of that falsy operand; it does not evaluate any of the remaining operands.

Consider the pseudocode below.

(some falsy expression) && expr

The expr part is never evaluated because the first operand (some falsy expression) is evaluated as [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy). 
```

...which you already knew 😄 Thanks!

Reply:
Yeah! This short circuit is very handy trick 🙌